libraries/helpers.rb

```diff
@@ -14,104 +14,103 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# Get the plan directory from the config, or fallback to /src, set
-# this in .delivery/config.json.
-#
-# Examples:
-# {
-#   ...
-#   "habitat": {
-#     "plan_dir": "/src/plans"
-#     "plan_dir": "/elsewhere/plans/myplan"
-#   }
-# }
-
-# The current acceptance environment on the server
-def existing_acceptance_environment
-  Chef::ServerAPI.new.get("environments/#{get_acceptance_environment}")
-rescue Net::HTTPServerException => e
-  raise e unless e.response.code.to_i == 404
-  {}
-end
-
-def habitat_plan_dir
-  if node['delivery']['config'].attribute?('habitat') &&
-     node['delivery']['config']['habitat'].attribute?('plan_dir')
-    node['delivery']['config']['habitat']['plan_dir']
-  else
-    '/src/habitat'
-  end
-end
-
-def habitat_origin_key?
-  project_secrets_exist?(%w(keyname private_key public_key))
-end
-
-def habitat_depot_token?
-  project_secrets_exist?(%w(depot_token))
-end
 
-def hab_binary
-  if platform_family?('mac_os_x')
-    '/usr/local/bin/hab'
-  else
-    '/bin/hab'
+module HabitatBuildCookbook
+  module Helpers
+    #
+    # Get the plan directory from the config, or fallback to /src, set
+    # this in .delivery/config.json.
+    #
+    # Examples:
+    # {
+    #   ...
+    #   "habitat": {
+    #     "plan_dir": "/src/plans"
+    #     "plan_dir": "/elsewhere/plans/myplan"
+    #   }
+    # }
+    def habitat_plan_dir
+      if node['delivery']['config'].attribute?('habitat') &&
+         node['delivery']['config']['habitat'].attribute?('plan_dir')
+        node['delivery']['config']['habitat']['plan_dir']
+      else
+        '/src/habitat'
+      end
+    end
+
+    def habitat_origin_key?
+      project_secrets_exist?(%w(keyname private_key public_key))
+    end
+
+    def habitat_depot_token?
+      project_secrets_exist?(%w(depot_token))
+    end
+
+    def hab_binary
+      if platform_family?('mac_os_x')
+        '/usr/local/bin/hab'
+      else
+        '/bin/hab'
+      end
+    end
+
+    # For example, if the project is `surprise-sandwich`, and we're in the
+    # Build stage's Publish phase, the slug will be:
+    #
+    #    `surprise-sandwich-build-publish`
+    #
+    def hab_studio_slug
+      [
+        node['delivery']['change']['project'],
+        node['delivery']['change']['stage'],
+        node['delivery']['change']['phase'],
+      ].join('-')
+    end
+
+    def hab_studio_path
+      File.join('/hab/studios', hab_studio_slug)
+    end
+
+    def changed_habitat_files?
+      # `changed_files` comes from the delivery-sugar DSL: https://git.io/vXvAm
+      changed_files.select { |changed_file| changed_file =~ %r{habitat/} }.any?
+    end
+
+    private
+
+    # if we're going to load secrets, we need to make sure we actually
+    # have the data!
+    def project_secrets_exist?(secret_keys = [])
+      load_delivery_chef_config
```

Comment:
Be sure this is what you want to do. This will cause the rest of the phase to execute against the Delivery Chef Server. If this is only a temporary connection to fetch some data, try using [`with_server_config`](https://github.com/chef-cookbooks/delivery-sugar/blob/master/README.md#running-against-the-automate-chef-server)